datajoint/dependencies.py

```diff
@@ -43,15 +43,15 @@ def load(self):
             self.add_node(n, primary_key=pk)
 
         # load foreign keys
-        keys = self._conn.query("""
+        keys = [{k.lower(): v for k, v in elem.items()} for elem in self._conn.query("""
```

Comment:
change list to generator expression for lazy evaluation.

Reply:
Good point, I have updated it to be a proper generator